### PR TITLE
UI: HDS Adoption upgrade to design-system-components: 3.3.0 and ember-flight-icons: 4.0.4

### PR DESCRIPTION
**Pull request by hellobontempo**
This PR is a prerequisite to implementing `Hds::Breadcrumbs`
   - upgrade `@hashicorp/design-system-components` to `^3.3.0` (`3.0.0` added linking to external routes)
   - upgrade  `@hashicorp/ember-flight-icons` to  `^4.0.4`

`3.0.0` introduced breaking changes. We were already using `bottom-left` and `bottom-right` so now these values will behave as expected. In addition, `data-clipboard-text` was removed as an attribute for the `Hds::CopyButton` and so tests had to be updated accordingly

> As part of this release, we’ve made the following major changes:
Added a visibility toggle to password input fields. This feature is enabled by default and can be disabled using the @hasVisibilityToggle argument.
Removed @listPosition’s `left` and `right` values in the Dropdown component. These values have been deprecated in [version 2.0.0](https://github.com/hashicorp/design-system/releases/tag/%40hashicorp%2Fdesign-system-components%402.0.0) and replaced with `bottom-left` and `bottom-right`, respectively.
Repurposed the @layout property in Radio Card Group to set the direction of the cards within the group. The previous values for this property were `fixed` (requiring a @maxWidth value to be specified to constrain the card) and `fluid` (default). Now a card becomes automatically fixed when a @maxWidth value is set. The repurposed @layout property can now be set to `vertical` or `horizontal` (default).
Removed the @layout property in the RadioCard component. Similar to the above, the card becomes fixed when a @maxWidth value is set.
Renamed `extraBefore` and `extraAfter` generic containers in the SideNav component to `ExtraBefore` and `ExtraAfter`, respectively, aligning them with the naming conventions.
Dropped support for Node 14 as it reached end-of-life in April 2023.

Note: _branch name isn't accurate because originally I was planning to replace the crumbs in this PR, but the upgrade blew up scope_

### CODE REVIEW

File: ui/app/components/policy-form.hbs [hunk 1]
```diff
@@ -60,7 +60,7 @@
               @isIconOnly={{true}}
               @textToCopy={{@model.policy}}
               class="transparent"
-              data-test-copy-button
```
Comment (hellobontempo) [inline ui/app/components/policy-form.hbs:63]: upgrading to `3.1.0` removed the test selector `data-test-clipboard` from the copy button - adding the clipboard value to the selector here is the easiest way to address test failures

File: ui/app/styles/helper-classes/layout.scss [hunk 2]
```diff
@@ -68,7 +68,7 @@
 }
 
 .is-medium-width {
-  width: $desktop / 3;
+  width: calc($desktop / 3);
```
Comment (hellobontempo) [inline ui/app/styles/helper-classes/layout.scss:71]: this class is used in the `certificate-card.hbs` so decided to tackle this deprecation warning while here
Screenshot 2023-12-04 at 3 47 00 PM

File: ui/lib/core/addon/components/certificate-card.hbs [hunk 3]
```diff
@@ -20,13 +20,13 @@
       {{@data}}
     </code>
   </div>
-  <div class="is-flex-center has-background-white-bis has-side-padding-s has-top-bottom-margin-negative-m">
+  <div class="is-flex-center has-background-white-bis has-top-bottom-margin-negative-m">
```
Comment (hellobontempo) [inline ui/lib/core/addon/components/certificate-card.hbs:23]: It looks like some styling changes to this component caused weird button padding here
# after upgrading to `3.1.0`
Screenshot 2023-12-04 at 3 48 10 PM

# addressed by removing side badding and now:
Screenshot 2023-12-04 at 3 47 22 PM

File: ui/yarn.lock
Comment (hellobontempo) [inline ui/yarn.lock:1]: @alex-ju - It seems like this file added quite a few dependencies when upgrading to `3.3.0`, could you take a look and confirm these look right to you?

Comment (hellobontempo) [inline ui/yarn.lock:1]: @alex-ju - It seems like this file added quite a few dependencies when upgrading to `3.3.0`. We jumped quite a few versions (from `2.13.0`), so it's not much of a surprise, but could you take a scan and confirm these look right to you? 👀

Comment (hellobontempo) [reply ui/yarn.lock:1]: Thanks for getting back to me! @didoo - not a huge rush here but if you have a moment to look over this that'd be helpful so I can update our breadcrumbs

File: ui/lib/core/addon/components/certificate-card.hbs [hunk 4]
```diff
@@ -20,13 +20,13 @@
...
     <Hds::Copy::Button
       @text="Copy"
       @isIconOnly={{true}}
       @textToCopy={{@data}}
-      class="transparent is-paddingless"
-      data-test-copy-button
+      class="transparent"
```
Comment (didoo) [inline ui/lib/core/addon/components/certificate-card.hbs:28]: what is this class doing? if it's removing the background and border from the copy button, make sure:
- the other background is differentiated for the different states (hover/active)
- there is a transparent border applied to the "button" emulation
- the focus is correctly applied (somewhere)

File: ui/app/components/sidebar/user-menu.hbs [hunk 5]
```diff
@@ -54,6 +54,7 @@
                 @container="#container"
                 class="in-dropdown link is-flex-start"
                 {{on "click" (fn (set-flash-message "Token copied!"))}}
+                data-test-clipboard-value={{this.auth.currentToken}}
```
Comment (didoo) [inline ui/app/components/sidebar/user-menu.hbs:57]: nitpicking, but it's odd to see so many variations of the same entity/name: `data-test-copy-button`, `data-test-clipboard-text`, `data-test-tooltip-copy`, `data-test-clipboard-value` 😐

**Review by hashishaw — APPROVED** [review]
LGTM

File: ui/yarn.lock
Comment (didoo) [reply ui/yarn.lock:1]: @hellobontempo there are too many changes in this lockfile to give any meaningful feedback 🙃 so I am going to approve trusting the fact that someone (you?) have tested that the application still works as expected (and all the tests are passing).

**Review by didoo — APPROVED** [review]
@hellobontempo I've added a couple of (non blocking) comments and I am approving (just in case you need a thumbs up).

File: ui/yarn.lock
Comment (hellobontempo) [reply ui/yarn.lock:1]: Yep! Sorry - it ended up being way more than I realized. Thanks for taking a look!

File: ui/lib/core/addon/components/certificate-card.hbs (hunk 4 above)
Comment (hellobontempo) [reply ui/lib/core/addon/components/certificate-card.hbs:28]: it's to stylize the copy button the same as `@color="tertiary"` to match the other buttons here
transparent

File: ui/app/components/sidebar/user-menu.hbs (hunk 5 above)
Comment (hellobontempo) [reply ui/app/components/sidebar/user-menu.hbs:57]: Yep - I iteratively replaced `data-clipboard-text` as I went through the tests - and purposefully differentiated the selector names to track which ones were updated. 

We don't have a consistent way of adding test selectors on our team, though we've _started_ moving towards using a general selector file so we can reuse selectors

File: ui/yarn.lock
Comment (alex-ju) [reply ui/yarn.lock:1]: all looks good! thanks for tackling this @hellobontempo!

**Pull request merged**